interpreter/CMakeLists.txt

```diff
@@ -80,7 +99,14 @@ set(LLVM_TARGETS_TO_BUILD ${ROOT_CLING_TARGET} CACHE STRING "Semicolon-separated
 if(clingtest)
   message("-- cling test suite enabled: llvm / clang symbols in libCling will be visible!")
   set(CLING_INCLUDE_TESTS ON CACHE BOOL "" FORCE)
-  ROOT_ADD_TEST(clingtest-check-cling COMMAND ${CMAKE_COMMAND} --build ${CMAKE_BINARY_DIR} --target check-cling)
+  # The path to cling passed through environment variable only relevant when building
+  # against external llvm. In that case, cling's testsuite cannot deduce the binary
+  # of cling relatively to the llvm tools folder.
+  if (NOT builtin_llvm)
+    set(CLINGTEST_EXECUTABLE CLING=${CMAKE_CURRENT_BINARY_DIR}/llvm/src/${CMAKE_CFG_INTDIR}/bin/cling)
+  endif()
+  ROOT_ADD_TEST(clingtest-check-cling COMMAND ${CMAKE_COMMAND} --build ${CMAKE_BINARY_DIR} --target check-cling
+                                       ENVIRONMENT ${CLINGTEST_EXECUTABLE})
```

Comment:
Just curiosity, but why is this needed? Could you use `$<TARGET_FILE:cling>` or similar in `check-cling` target to avoid this?

Reply:
We did not need this before, because we knew where to find cling, relative to the `llvm_tools_dir`. Having builtin_llvm=Off sets that variable to something else and we cannot find back the cling binary.

Of course we can probably find a hackish solution but I did not find anything that I liked. Currently the `check-cling` target is only available when configuring with -Dclingtest=On and if we have -Dbuiltin_llvm=Off we will need to provide the path to the cling binary if we want to dial the target outside of roottest.